File: Hard/2026-07-03-3620-Network-Recovery-Pathways/solution.js
class Solution {
    maxScore(edges, online, k) {
        const n = online.length;
        const graph = Array.from({ length: n }, () => []);
        
        for (const [u, v, cost] of edges) {
            graph[u].push({ v, cost });
        }
        
        const bfs = (minEdgeCost) => {
            const queue = [{ node: 0, totalCost: 0 }];
            const visited = new Set();
            visited.add(0);
            
            while (queue.length) {
                const { node, totalCost } = queue.shift();
                
                if (node === n - 1) return true;
                
                for (const { v, cost } of graph[node]) {
                    if (online[v] && !visited.has(v) && totalCost + cost <= k && cost >= minEdgeCost) {
                        visited.add(v);
                        queue.push({ node: v, totalCost: totalCost + cost });
                    }
                }
            }
            return false;
        };
        
        let left = 0, right = Math.max(...edges.map(edge => edge[2])), result = -1;
        
        while (left <= right) {
            const mid = Math.floor((left + right) / 2);
            if (bfs(mid)) {
                result = mid;
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
        
        return result;
    }
}

File: Hard/2026-07-03-3620-Network-Recovery-Pathways/solution.java
import java.util.*;

public class Solution {
    public int maxScore(int[][] edges, boolean[] online, long k) {
        int n = online.length;
        List<List<int[]>> graph = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            graph.add(new ArrayList<>());
        }
        
        for (int[] edge : edges) {
            graph.get(edge[0]).add(new int[]{edge[1], edge[2]});
        }
        
        int left = 0, right = Integer.MAX_VALUE, result = -1;
        
        while (left <= right) {
            int mid = left + (right - left) / 2;
            if (canReach(graph, online, k, mid)) {
                result = mid;
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
        
        return result;
    }
    
    private boolean canReach(List<List<int[]>> graph, boolean[] online, long k, int minCost) {
        int n = online.length;
        long[] dist = new long[n];
        Arrays.fill(dist, Long.MAX_VALUE);
        dist[0] = 0;
        
        Queue<Integer> queue = new LinkedList<>();
        queue.offer(0);
        
        while (!queue.isEmpty()) {
            int node = queue.poll();
            if (!online[node]) continue;
            
            for (int[] edge : graph.get(node)) {
                int neighbor = edge[0];
                int cost = edge[1];
                
                if (cost < minCost || !online[neighbor]) continue;
                
                if (dist[node] + cost <= k && dist[node] + cost < dist[neighbor]) {
                    dist[neighbor] = dist[node] + cost;
                    queue.offer(neighbor);
                }
            }
        }
        
        return dist[n - 1] != Long.MAX_VALUE;
    }
}